packages/blade/src/components/List/ListContext.ts

```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import type { IconComponent } from '../Icons';
+import type { ListProps } from './List';
+
+export type ListContextType = {
+  level?: number;
+  size: NonNullable<ListProps['size']>;
+  icon?: IconComponent;
+  variant: NonNullable<ListProps['variant']>;
+};
+
+const ListContext = React.createContext<ListContextType>({
+  level: undefined,
+  size: 'medium',
+  variant: 'unordered',
+});
+const ListProvider = ListContext.Provider;
+
+const useListContext = (): ListContextType => {
+  const context = React.useContext(ListContext);
+  return context;
```

Comment:
Let's throw an error here if `ListItem` is used outside of <List \/>?

if(!context) throw error